spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AzureCloudConfigWatch.java

```diff
@@ -107,28 +108,57 @@ private boolean needRefresh(ConfigStore store) {
                 .withLabels(store.getLabels()).withFields(QueryField.ETAG).withRange(0, 0);
 
         List<KeyValueItem> keyValueItems = configOperations.getRevisions(store.getName(), options);
+        
         if (keyValueItems.isEmpty()) {
             return false;
         }
 
         String etag = keyValueItems.get(0).getEtag();
-        if (firstTimeMap.get(store.getName()) == null) {
-            storeEtagMap.put(store.getName(), etag);
-            firstTimeMap.put(store.getName(), false);
+        if (firstTimeMap.get(store.getName() + "configuration") == null) {
```

Comment:
How about make this `configuration` literal to constant? Same with the `feature` literal. Also, how about add a underscore before this, e.g., `_configuration`?